wurst/systems/spells/mage/Depress.wurst

```diff
@@ -1,15 +1,50 @@
 package Depress
 import RegisterEvents
 import ClosureTimers
+import AbilityObjEditing
+import ObjectIdGenerator
+import BuffObjEditing
+import ObjectIds
+import Orders
+import IttDummyCaster
 
 @configurable constant int ABILITY_ID = 'A025'
+let BUFF_ID = compiletime(BUFF_ID_GEN.next())
+let DUMMY_SLOW_ID = compiletime(ABIL_ID_GEN.next())
+let MS_SLOW = 0.2
+let AS_SLOW = 0.1
+
+@compiletime function createBuffs()
+    new BuffDefinition(BUFF_ID, 'Bfro')
+        ..setTooltipNormal(1, "Depress Slow")
+        ..setTooltipNormalExtended(1, "Slowed by Depress")
+        ..setIconNormal(1, "ReplaceableTextures\\CommandButtons\\BTNShadowPact.blp")
+        ..setIcon("ReplaceableTextures\\CommandButtons\\BTNShadowPact.blp")
+        ..setArtTarget(1, "")
+
+@compiletime function createSlowDummySpell()
+    new AbilityDefinitionSlowCreep(DUMMY_SLOW_ID)
+        ..setAttackSpeedFactor(1, AS_SLOW)
+        ..setMovementSpeedFactor(1, MS_SLOW)
+        ..setBuffs(1, toRawCode(BUFF_ID))
+        ..setCastRange(1, 2500)
+        ..setCastingTime(1, 0)
+        ..setCooldown(1, 0)
+        ..setDurationHero(1, 20)
+        ..setDurationNormal(1, 20)
+        ..setManaCost(1, 0)
+        ..setTargetsAllowed(1, "air,allies,enemies,ground")
+        ..setName("Depress dummy slow")
 
 function onCast()
-    var target = GetSpellTargetUnit()
-    if target.getMana() < 40
-        return
-    target.subMana(10)
-    doPeriodicallyCounted(2.00, 4, (cb) -> target.subMana(5))
+    let target = GetSpellTargetUnit()
+    let caster = GetSpellAbilityUnit()
+    let owner = caster.getOwner()
+    doAfter(0.2) ->
```

Comment:
why did you add the delay?

Reply:
To approximate projectile flight time. Without a custom projectile I dont see a way to detect the impact otherwise.

Reply:
That doesnt differentiate between the exact ability that deals the damage though. And also a lot of effort for just a simple effect like this to have (even more) correct timing

Reply:
You’d have to cast it from a dummy then detect if the dummy was the damage source

Reply:
I agree it’s unnecessary though